packages/plugin-node/src/services/imageUtils.ts

```diff
@@ -0,0 +1,79 @@
+import sharp from "sharp";
+import * as FileType from "file-type/core";
+
+interface ImageDimensions {
+    width: number;
+    height: number;
+}
+
+interface ProcessedImage {
+    buffer: Buffer;
+    mimeType: string;
+    dimensions: {
+        original: ImageDimensions;
+        resized: ImageDimensions;
+    };
+}
+
+export async function resizeImageBuffer(
+    imageBuffer: Buffer,
+    maxWidth: number,
+    maxHeight: number
+): Promise<ProcessedImage> {
+    // Detect MIME type
+    try {
+        // Detect MIME type
+        const fileTypeResult = await FileType.fileTypeFromBuffer(imageBuffer);
+        if (!fileTypeResult || !fileTypeResult.mime.startsWith("image/")) {
+            throw new Error("Invalid image format");
+        }
+
+        // Get original image metadata
+        const metadata = await sharp(imageBuffer).metadata();
+        if (!metadata.width || !metadata.height) {
+            throw new Error("Could not get image dimensions");
+        }
+
+        // Calculate new dimensions maintaining aspect ratio
+        let width = metadata.width;
+        let height = metadata.height;
+
+        if (width > maxWidth) {
+            height = Math.round((maxWidth * height) / width);
+            width = maxWidth;
+        }
+
+        if (height > maxHeight) {
+            width = Math.round((maxHeight * width) / height);
+            height = maxHeight;
+        }
+
+        // Process the image
+        const resizedBuffer = await sharp(imageBuffer)
+            .resize(width, height, {
+                fit: "inside",
+                withoutEnlargement: true,
+            })
+            .toBuffer();
```

Comment:
The image processing could lead to performance issues if the input image is large. Consider implementing a size limit check before processing to avoid excessive memory usage.